values/conditional.gen.go.tmpl

```diff
@@ -22,71 +22,89 @@ import (
 //
 // Cases where either or both of the consequent `c` and alternate `a` are
 // `VectorRepeatValue`s are supported.
-func VectorConditional(t, c, a Vector, mem memory.Allocator) (Value, error) {
-	var tvr, cvr, avr *Value
-	if vr, ok := t.(*VectorRepeatValue); ok {
-		tvr = &vr.val
-	}
-	if vr, ok := c.(*VectorRepeatValue); ok {
-		cvr = &vr.val
-	}
-	if vr, ok := a.(*VectorRepeatValue); ok {
-		avr = &vr.val
+//
+// Similar to the vec repeat situation, cases where either `c` or `a` are
+// invalid (aka null), we treat this as if we have a null constant for the given
+// branch. Items selected from the branch will produce nulls in the output.
+// For cases where both `c` and `a` are null, we short circuit immediately
+// returning a new untyped null Value.
+func VectorConditional(t Vector, c, a Value, mem memory.Allocator) (Value, error) {
+	// We will know the output type only when at least one branch is not null.
+	// When both branches are null, the output can only be null (aka Invalid).
+	var elemType semantic.MonoType
+	if c.IsNull() && a.IsNull() {
+		return New(nil), nil // early return an untyped null
```

Comment:
I did notice there's a sentinel value, `values.Null`, which could be used here instead of going through the trouble of calling `New()`. Spotted this while I was auditing the code path on the binary side (for the recent equality ops PR).

The similar case for binary ops is handled here

https://github.com/influxdata/flux/blob/32fe18dbcdc6ea4a30cf467231c84cdbf8d1bd54/values/binary.go#L55-L62